tests/cromwellapi/conftest.py

```diff
@@ -24,16 +65,66 @@ def submit_wdls(cromwell_api):
     pattern = "**/*.wdl"
     wdl_paths = list(root.glob(pattern))
 
-    print(f"Submitting {len(wdl_paths)} wdls ...")
-
-    out = []
-    for path in wdl_paths:
-        opts_path = path.parent / "options.json"
-        opts_path = opts_path if opts_path.exists() else None
-        res = cromwell_api.submit_workflow(wdl_path=path, options=opts_path)
-        out.append(res)
+    if recording_mode == "rewrite":
+        print(f"\nSubmitting {len(wdl_paths)} wdls ...")
+        out = []
+        for path in wdl_paths:
+            opts_path = path.parent / "options.json"
+            opts_path = opts_path if opts_path.exists() else None
+            res = cromwell_api.submit_workflow(wdl_path=path, options=opts_path)
+            out.append(res)
+        with open(mocked_submissions, "w") as f:
+            json.dump(out, f, indent=4)
+    else:
+        with open(mocked_submissions, "r") as f:
+            out = json.load(f)
 
     # Yield to let tests run
     yield out
 
     # Cleanup is not possible for Cromwell - but would be here
+
+
+class EnvironmentVariableError(Exception):
+    """Custom error class for a missing environment variable"""
+    pass
+
+
+def filter_response_bodies(response):
```

Comment:
smart and pretty simple solution, thank you

Reply:
👍🏽